[Changelog] Tunnel: UDP Performance & Reliability Improvements (#23684)
cloudflare/cloudflare-docs: src/content/changelog/cloudflare-tunnel/2025-07-15-udp-improvements.mdx

---
title: Faster, more reliable UDP traffic for Cloudflare Tunnel

> description: We've re-architected how Cloudflare Tunnel handles UDP traffic. Your real-time applications are now faster and more stable, especially on Tunnels with mixed TCP and UDP traffic.

Reviewer: ```suggestion
description: We've re-architected how Cloudflare Tunnel handles UDP traffic. Your real-time applications are now faster and more stable, especially on tunnels with mixed TCP and UDP traffic.
```

date: 2025-07-15
---

import { Render } from "~/components";

Your real-time applications running over [Cloudflare Tunnel](/cloudflare-one/connections/connect-networks/) are now faster and more reliable. We've completely re-architected the way `cloudflared` proxies UDP traffic in order to isolate it from other traffic, ensuring latency-sensitive applications like private DNS are no longer slowed down by heavy TCP traffic (like file transfers) on the same Tunnel.

> This is a foundational improvement to Cloudflare Tunnel, delivered automatically to all customers. There are no settings to configure — your UDP traffic is already flowing faster and more reliably.

Reviewer: ```suggestion
This is a foundational improvement to Cloudflare Tunnel, delivered automatically to all customers. You do not need to configure any settings or update `cloudflared` — your UDP traffic is already flowing faster and more reliably.
```
